examples/todo/src/task.rs

```diff
@@ -27,32 +28,39 @@ pub struct Todo {
 }
 
 impl Task {
-    pub fn all(conn: &SqliteConnection) -> Vec<Task> {
-        all_tasks.order(tasks::id.desc()).load::<Task>(conn).unwrap()
+    pub fn all(conn: &SqliteConnection) -> Result<Vec<Task>> {
+        all_tasks.order(tasks::id.desc()).load::<Task>(conn)
     }
 
-    pub fn insert(todo: Todo, conn: &SqliteConnection) -> bool {
+    pub fn insert(todo: Todo, conn: &SqliteConnection) -> Result<()> {
         let t = Task { id: None, description: todo.description, completed: false };
-        diesel::insert_into(tasks::table).values(&t).execute(conn).is_ok()
+        diesel::insert_into(tasks::table).values(&t).execute(conn)?;
+
+        Ok(())
     }
 
-    pub fn toggle_with_id(id: i32, conn: &SqliteConnection) -> bool {
-        let task = all_tasks.find(id).get_result::<Task>(conn);
-        if task.is_err() {
-            return false;
-        }
+    pub fn toggle_with_id(id: i32, conn: &SqliteConnection) -> Result<()> {
+        let task = all_tasks.find(id).get_result::<Task>(conn)?;
 
-        let new_status = !task.unwrap().completed;
+        let new_status = !task.completed;
         let updated_task = diesel::update(all_tasks.find(id));
-        updated_task.set(task_completed.eq(new_status)).execute(conn).is_ok()
+        updated_task.set(task_completed.eq(new_status)).execute(conn)?;
+
+        Ok(())
     }
 
-    pub fn delete_with_id(id: i32, conn: &SqliteConnection) -> bool {
-        diesel::delete(all_tasks.find(id)).execute(conn).is_ok()
+    pub fn delete_with_id(id: i32, conn: &SqliteConnection) -> Result<()> {
+        match diesel::delete(all_tasks.find(id)).execute(conn) {
```

Comment:
This is just:

```suggestion
        diesel::delete(all_tasks.find(id)).execute(conn).map(|_| ())
```

What's the type of the `Ok`, and, should we just throw it away? Is there any utility to returning it?

Reply:
https://docs.rs/diesel/1.4.3/diesel/query_dsl/trait.RunQueryDsl.html#method.execute

Evidently this returns the number of rows affected.
Some sort of behind the scenes database modification metrics could be saved potentially, but I think for an example it's best to leave that kind of functionality out and simply leave a comment explaining that the number of rows changed are returned for those who want them?

Reply:
Map was being used here, but Jebrosen indicated in some other places that he found `matches` more readable in general, so I changed here as well for consistency. In retrospect, I probably should've evaluated this on a case by case basis instead of trying to achieve all around consistency.